### ISSUE TITLE
Ageing Pages report causes KeyError due to deleted user

### ISSUE THREAD

**Issue by jams2** [wagtail/wagtail#9520]
### Issue Summary

When rendering the Ageing Pages report, if a page in the query set was most recently published by a user that has subsequently been deleted, a `KeyError` is thrown, displaying an error page to the user.

Traceback:

```
Internal Server Error: /admin/reports/aging-pages/
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/decorators/cache.py", line 62, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/code/wagtail/wagtail/admin/urls/__init__.py", line 165, in wrapper
    return view_func(request, *args, **kwargs)
  File "/code/wagtail/wagtail/admin/auth.py", line 182, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/code/wagtail/wagtail/admin/views/generic/permissions.py", line 36, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
    return handler(request, *args, **kwargs)
  File "/code/wagtail/wagtail/admin/views/reports/base.py", line 30, in get
    context["object_list"] = self.decorate_paginated_queryset(
  File "/code/wagtail/wagtail/admin/views/reports/aging_pages.py", line 71, in decorate_paginated_queryset
    page.last_published_by_user = username_mapping[page.last_published_by]
KeyError: 4
```
Encountered on 3.0.3, duplicated on 4.0.4.

It is caused by the dict lookup in [this function](https://github.com/wagtail/wagtail/blob/78a7652f6b6495c6e145cf7aa0774f4b0690b6f0/wagtail/admin/views/reports/aging_pages.py#L61)

### Steps to Reproduce

From a working Wagtail application, e.g. `bakerydemo`:

1. As an admin user, create an additional user with editing permissions
2. Log out, log back in as that new user
3. Edit and publish a page
4. Confirm it appears in the ageing pages report, attributed to your new user
5. Log out, log back in as admin
6. Delete the newly created user
7. Visit the ageing pages report

Suggested solution: check for the existence of the user in the `username_mapping` dict, and render the string "(user not found)" instead of the username - or "user {id} (deleted)" like the page history view - if the user's id is not in the dict.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.8.12
-   Django version: 4.0.8
-   Wagtail version: 4.0.4
-   Browser version: Firefox 106

**Comment by nickmoreton** [comment]
I can confirm this issue is present in 4.0.4 following an upgrade of an existing client site.

```
Internal Server Error: /admin/reports/aging-pages/
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/venv/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/wagtail/admin/urls/__init__.py", line 165, in wrapper
    return view_func(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/wagtail/admin/auth.py", line 182, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/wagtail/admin/views/generic/permissions.py", line 36, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/venv/lib/python3.8/site-packages/wagtail/admin/views/reports/base.py", line 30, in get
    context["object_list"] = self.decorate_paginated_queryset(
  File "/venv/lib/python3.8/site-packages/wagtail/admin/views/reports/aging_pages.py", line 71, in decorate_paginated_queryset
    page.last_published_by_user = username_mapping[page.last_published_by]
KeyError: 8
```

I have no user with the ID 8

Python version: 3.8.15
Django version: 3.2.16
Wagtail version: 4.0.4

**Comment by gasman** [comment]
Fixed in #9531.
